### ISSUE TITLE
A few broken builds

### ISSUE THREAD

**Issue by arokem** [nipy/dipy#1138]
With errors of this kind: 

http://nipy.bic.berkeley.edu/builders/dipy-bdist32-27/builds/378/steps/shell_9/logs/stdio
http://nipy.bic.berkeley.edu/builders/dipy-bdist32-33/builds/283/steps/shell_9/logs/stdio
http://nipy.bic.berkeley.edu/builders/dipy-bdist32-34/builds/287/steps/shell_9/logs/stdio

I *think* that they originated from this PR: https://github.com/nipy/dipy/pull/1092. 

@nilgoyyou, @Garyfallidis : could you please take a look and confirm?

**Comment by nilgoyyou** [comment]
I think it's related to the problem described [here](http://cython.readthedocs.io/en/latest/src/tutorial/external.html#dynamic-linking): "The libc math library is special in that it is not linked by default on some Unix-like systems, such as Linux."

I reproduced the problem in Windows.
- I tried adding the `libraries=["m"]` parameter but `m.lib` doesn't exist on Windows. Or `math.lib`, or `cmath.lib`, etc. I don't really know what to write here.
- I tried redeclaring `fmin` with `cdef extern from "math.h" nogil:` but it stays undefined. Anyway, we can [clearly see that it's defined in cython](https://github.com/cython/cython/blob/master/Cython/Includes/libc/math.pxd) so we shouldn't have to do that.

This being said, I was not able to fix it on my computer. Anyone known how?

**Comment by matthew-brett** [comment]
This may be relevant : http://stackoverflow.com/questions/19234452/visual-c-express-fmin-and-fmax-identifier-not-found#19234517

```
fmin and fmax are a C99 feature. Visual Studio only implements the C89 standard, so it doesn't have that functions.
```

Maybe you could do something like [this](http://powdertoy.co.uk/Discussions/Thread/View.html?Thread=8877&PageNum=0#Message=140656) in `src/dpy_math.h`:

```
#if (defined(WIN32) || defined(_WIN64)) && !defined(__GNUC__)
#define fmin min
#endif
```

Then in the pyx file:

```
cdef extern from "dpy_math.h" nogil:
    double fmin(double x)
```

or similar.  @omarocegueda may better remember our struggles on this kind of thing back in the day.

**Comment by arokem** [comment]
This is addressed in #1151
